contract_payment_mode/models/contract.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    payment_mode_id = fields.Many2one(
+        comodel_name='account.payment.mode',
+        string='Payment Mode',
+        domain=[('payment_type', '=', 'inbound')],
+    )
+
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        if self.partner_id.customer_payment_mode_id:
+            self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+
+    @api.model
+    def _prepare_invoice_data(self, contract):
```

Comment:
As we have seen on v9, this method is incorrect. Correct one is `_prepare_invoice` (see https://github.com/OCA/contract/pull/73/files#diff-237fb5082150850f78bb85c3c23bf563R20)